django/contrib/postgres/fields/array.py

```diff
@@ -32,6 +34,8 @@ def __init__(self, base_field, size=None, **kwargs):
         # implements it.
         if hasattr(self.base_field, 'from_db_value'):
             self.from_db_value = self._from_db_value
+        if isinstance(self.base_field, GeometryField):
```

Comment:
I think you'll want to define `GeometryField.from_db_value` instead.

Reply:
The value which is returned from for example `ArrayField(PointField)` is a string like:
`{HexRepresentationofPoint1,HexRepresentationofPoint2}`

So, if I define `GeometryField.from_db_value`, this method will receive the above string character by character because of this [block](https://github.com/django/django/blob/7b31ba541f1dfb3a8e782b1319c25a24f9d86f8a/django/contrib/postgres/fields/array.py#L121:L124)

That's why I did it in this way

Reply:
I think this will be a no-go because it couples `contrib.gis` with `contrib.postgres` though. See [#28161](https://code.djangoproject.com/ticket/28161) which added support for `ArrayField(CITextField)`. Looks like we need to do something similar for `gis` fields when the application is installed. That's starting to be a lot of things that are run a connection creation though.

Reply:
I agree with Simon, we need to register a new array type.

Reply:
Thanks @charettes for your suggestion.
I've tried to create and register a new array type for `geometry` based on your suggestion at #8453. here is the code:
```
def get_geometry_oids(connection_alias):
    with connections[connection_alias].cursor() as cursor:
        cursor.execute("SELECT typarray FROM pg_type WHERE typname = 'geometry'")
        return tuple(row[0] for row in cursor)

...

       geometry_oids = get_geometry_oids(connection.alias)
        new_type = psycopg2.extensions.new_type(geometry_oids, "geometry", caster) # caster is function to cast hex to geometry object
        array_type = psycopg2.extensions.new_array_type(geometry_oids, 'geometry[]', new_type)
        psycopg2.extensions.register_type(array_type, None)
```
The problem is that in `caster` function, I get all items in array not one item. something like this:
`0101000020E610000000000000000018400000000000003840:0101000020E610000000000000000014400000000000003740`
which is represent two points that seprated by `:`.
I've checed the column value in postgres console. for an array of points I got this:
`{0101000020E6100000000000000000F03F0000000000000040:0101000020E6100000000000000000F03F0000000000000040}`
While, I got below value for an array of chars:
`{foo,bar}`

the difference between two cases is for an array of points I got `:` as a separator,  but for an array of chars I got `,`.

I also tried to insert data from console to an array of points column by:
`insert into test values (ARRAY[st_geomFromText('POINT(1 1)', 4326), st_geomFromText('POINT(1 1)', 4326)]);`
the result was the same(`:` as separator).

Do you have any idea?